django/contrib/auth/tokens.py

```diff
@@ -12,7 +12,12 @@ class PasswordResetTokenGenerator:
     """
     key_salt = "django.contrib.auth.tokens.PasswordResetTokenGenerator"
     algorithm = 'sha256'
-    secret = settings.SECRET_KEY
+    secret = None
+
+    def __init__(self):
+        # Defer reading of the SECRET_KEY to runtime, otherwise this will
+        # raise ImproperlyConfigured during runserver startup.
```

Comment:
I'm not sure you need this comment. Import time reading of settings is generally known to be bad since it prevents e.g. `override_settings` from working (but you could add a test that uses `PasswordResetTokenGenerator` with `PasswordResetTokenGenerator`)